### PR TITLE
Add grid order arg fix column major calculation

### PR DESCRIPTION
**Pull request by Damon3000s**
- Add GridOrder to the API to control the order that items are shown in
  - RowMajor -> left to right priority
  - ColumnMajor -> top to bottom priority
- Tweaked the demo for more flexibility. Instead of adding new grid sections to show a different number of items, instead add a slider that will dynamically change how many items are displayed in the grid. Useful for testing and debugging various different scenarios with different icon counts and window sizes.

### CODE REVIEW

**Comment by matt-edmondson** [comment]
Is this taking my existing hidden implementation (bool: columnFirst) and productionifying it?

The diff was kinds hard to follow due to reordering, but I think the main point of difference is lines 125-152. 

Can you explain a bit why its needed over what the original implementation was doing?

File: ImGuiWidgets/Grid.cs [hunk 1]
```diff
@@ -103,6 +122,34 @@ public static void Show<T>(IEnumerable<T> items, MeasureGridCell<T> measureDeleg
...
+				{
+					List<int> itemsPerRow = [];
+					int maxItemsPerRow = numColumns;
+					int minItemsPerRow = numColumns - 1;
+
+					int count = 0;
+					for (int i = 0; i < itemList.Length; i++)
+					{
+						count++;
+						if ((i % numColumns == 0) && (i != itemList.Length))
```
Comment (Damon3000s) [inline ImGuiWidgets/Grid.cs:140]: Make locals and name better

File: ImGuiWidgets/Grid.cs [hunk 2]
```diff
@@ -8,6 +8,21 @@ namespace ktsu.ImGuiWidgets;
 /// </summary>
 public static partial class ImGuiWidgets
 {
+	public enum GridOrder
+	{
+		// Items are displayed in order left to right before dropping to the next row
+		// Recommended for when displaying grids of icons
+		// [1] [2] [3]
+		// [4] [5] [6]
+		RowMajor,
```
Comment (Damon3000s) [inline ImGuiWidgets/Grid.cs:17]: Include another example with uneven display

File: ImGuiWidgets/Grid.cs [hunk 3]
```diff
@@ -8,6 +8,21 @@ namespace ktsu.ImGuiWidgets;
...
+		// Items are displayed top to bottom before moving to the next column
+		// Note: Only 1 row may show 1 missing item. All other rows will be full.
+		// Recommended when displaying tables of data
+		// [1] [3] [5]
+		// [2] [4] [6]
+		ColumnMajor,
```
Comment (Damon3000s) [inline ImGuiWidgets/Grid.cs:23]: Include another example with a different display (uneven)

File: ImGuiWidgets/Grid.cs (hunk 1 above)
Comment (Damon3000s) [reply ImGuiWidgets/Grid.cs:140]: Done

File: ImGuiWidgets/Grid.cs (hunk 2 above)
Comment (Damon3000s) [reply ImGuiWidgets/Grid.cs:17]: Done

**Review by matt-edmondson — APPROVED** [review]
